pkg/scalers/arangodb_scaler.go

```diff
@@ -0,0 +1,276 @@
+	// +required
+	query string
+	// A threshold that is used as targetAverageValue in HPA.
+	// +required
+	queryValue int64
+	// A threshold that is used to check if scaler is active.
+	// +optional
+	activationQueryValue int64
+	// Specify whether to verify the server's certificate chain and host name.
+	// +optional
+	unsafeSsl bool
+	// Specify the max size of the active connection pool.
+	// +optional
+	connectionLimit int64
+
+	// The index of the scaler inside the ScaledObject
+	// +internal
+	scalerIndex int
+}
+
+// NewArangoDBScaler creates a new arangodbScaler
+func NewArangoDBScaler(config *ScalerConfig) (Scaler, error) {
+	metricType, err := GetMetricTargetType(config)
+	if err != nil {
+		return nil, fmt.Errorf("error getting scaler metric type: %s", err)
+	}
+
+	meta, err := parseArangoDBMetadata(config)
+	if err != nil {
+		return nil, fmt.Errorf("error parsing arangoDB metadata: %s", err)
+	}
+
+	client, err := getNewArangoDBClient(meta)
+	if err != nil {
+		return nil, err
+	}
+
+	return &arangoDBScaler{
+		metricType: metricType,
+		metadata:   meta,
+		client:     client,
+		logger:     InitializeLogger(config, "arangodb_scaler"),
+	}, nil
+}
+
+func getNewArangoDBClient(meta *arangoDBMetadata) (driver.Client, error) {
+	var auth driver.Authentication
+
+	conn, err := http.NewConnection(http.ConnectionConfig{
+		Endpoints: strings.Split(meta.endpoints, ","),
+		TLSConfig: &tls.Config{
+			MinVersion:         tls.VersionTLS13,
+			InsecureSkipVerify: meta.unsafeSsl,
+		},
+	})
+	if err != nil {
+		return nil, fmt.Errorf("failed to create a new http connection, %v", err)
+	}
+
+	if meta.arangoDBAuth.EnableBasicAuth {
+		auth = driver.BasicAuthentication(meta.arangoDBAuth.Username, meta.arangoDBAuth.Password)
+	} else if meta.arangoDBAuth.EnableBearerAuth {
+		hdr, err := jwt.CreateArangodJwtAuthorizationHeader(meta.arangoDBAuth.BearerToken, meta.serverID)
+		if err != nil {
+			return nil, fmt.Errorf("failed to create bearer token authorization header, %v", err)
+		}
+		auth = driver.RawAuthentication(hdr)
+	}
+
+	client, err := driver.NewClient(driver.ClientConfig{
+		Connection:     conn,
+		Authentication: auth,
+	})
+
+	if err != nil {
+		return nil, fmt.Errorf("failed to initialize a new client, %v", err)
+	}
+
+	return client, nil
+}
+
+func parseArangoDBMetadata(config *ScalerConfig) (*arangoDBMetadata, error) {
+	var err error
+
+	// setting default metadata
+	meta := arangoDBMetadata{}
+
+	// parse metaData from ScaledJob config
+	meta.endpoints, err = GetFromAuthOrMeta(config, "endpoints")
+	if err != nil {
+		return nil, err
+	}
+
+	if val, ok := config.TriggerMetadata["collection"]; ok {
+		meta.collection = val
+	} else {
+		return nil, fmt.Errorf("no collection given")
+	}
+
+	if val, ok := config.TriggerMetadata["query"]; ok {
+		meta.query = val
+	} else {
+		return nil, fmt.Errorf("no query given")
+	}
+
+	if val, ok := config.TriggerMetadata["queryValue"]; ok {
+		queryValue, err := strconv.ParseInt(val, 10, 64)
```

Comment:
Should we support `float` or arango always returns integers?

Reply:
We're counting the number of records for a given query, similar to MongoDB, hence it would always be an integer.

Reply:
Why limiting to counting? I mean, why can't we just execute the user given query and use that value, If a user wants only the row count, it's easy as executing the count function inside the query, isn't? Other database scalers (e.g mysql) just execute the given query and returns the value, giving the option to count rows or any other query. 
Maybe this is totally crazy in ArangoDB scaler context, but I have felt always that limiting the options to just counting records isn't good, having the users the option to count records just using the engine counting function.
WDYT?

Reply:
You're right. I have now updated the code accordingly.